.changes/unreleased/Added-20250723-184041.yaml

```diff
@@ -0,0 +1,13 @@
+kind: Added
+body: |-
+  Automatically apply .gitignore patterns on directory loading for module call 
+  
+  - Apply .gitignore patterns when loading contextual directory argument.
+  - Apply .gitignore pattenrs when a directory is sent as an argument to a module function.
+  - Apply .gitignore patterns when a local module is loaded.
+
+  Add `NoGitAutoIgnore` argument to `Host.Directory` to disable this behavior.
```

Comment:
This is a breaking change - calling `Host.directory` will now give different results. 

On the fence as to whether we want to do this or not. I think we definitely can break for context dirs and module loading, but we might want to preserve the behavior with `Host.directory`.

We'd do this by having `GitAutoIgnore` instead, and just setting it when loading.